static/js/protein/chart.ts

```diff
@@ -432,6 +460,280 @@ export function drawProteinInteractionChart(
     .on("mouseout", () => mouseout());
 }
 
+export function drawProteinInteractionGraph(
+  elementID: string,
+  data: InteractingProteinType[]
+): void {
+  /*
+  References:
+    1) Force-directed layout Observable: https://observablehq.com/@d3/force-directed-graph
+    2) Andrew Chen's force-directed layout with text labels tutorial: https://www.youtube.com/watch?v=1vHjMxe-4kI
+  */
+
+  const MAX_INTERACTIONS = 10;
+
+  const NODE_STYLE = {
+    circles: {
+      fillColors: ["mistyrose", "peachpuff", "lightCoral", "lightsalmon"],
+      radius: 15,
+      stroke: {
+        color: "#fff",
+        opacity: 1,
+        width: 1.5,
+      },
+    },
+    labels: {
+      font: {
+        color: "#222",
+        name: "public sans",
+        size: "8px",
+      },
+    },
+  };
+
+  const LINK_STYLE = {
+    length: 100,
+    stroke: {
+      color: "#999",
+      linecap: "round",
+      opacity: 0.6,
+      scoreWidthMultiplier: 8,
+    },
+  };
+
+  /*
+  DATA PROCESSING
+  Format post-processing:
+
+    nodes = [
+      { id: 1, name: "MECOM", species: "Human", depth: 0 },
+      { id: 2, name: "CtBP1", species: "Human", depth: 1 },
+      { id: 3, name: "SUPT16H", species: "Human", depth: 1 }
+    ]
+
+    links = [
+      { source: 1, target: 2, score: 0.3 },
+      { source: 1, target: 3, score: 0.7 }
+    ]
+  */
+
+  function nodeFromID(protein_speciesID: string, depth: number): ProteinNode {
+    // protein_speciesID: id of form {protein}_{species}, e.g. P53_HUMAN
+    const lastIndex = protein_speciesID.lastIndexOf("_"); // danger: assumes species name does not contain _
```

Comment:
Thanks will do! And will check with Sam about the underscores - I had used `lastIndexOf` because I thought it was less likely for species name to contain an underscore but I could definitely have this backwards.